mypy/plugins/attrs.py

```diff
@@ -60,19 +60,16 @@ class Converter:
     """Holds information about a `converter=` argument"""
 
     def __init__(self,
-                 type: Optional[Type] = None,
-                 is_attr_converters_optional: bool = False,
-                 is_invalid_converter: bool = False) -> None:
-        self.type = type
-        self.is_attr_converters_optional = is_attr_converters_optional
```

Comment:
Why don't we need these attributes any more?

Reply:
Because the `init_type` inference logic was moved from `Attribute` initialization to [`parse_converter()`](https://github.com/python/mypy/pull/12815/files#diff-c06d82e1af489188cdaa58e70e6460a858e2d4a993281e440a0c581947d69d81R592) and all of those attributes were only used for this inference. Now they are used internally on the parsing of the converter, but are not carried around in the `Converter` object, only the relevant `init_type`.
_________
In a lot more detail:
####
`Converter.type` was holding the type of the function that was used as a converter. Now that we want to also deal with converters that do not have such a type, this field would be less meaningful. As this type is only used to infer an `init_type` for the attribute, and we now already have the `init_type`, we do not need to keep this value.
`is_attr_converters_optional` was used before to take the (deserialized) `Converter` object and when inferring an `init_type` from it and making it that type a `Union` with `NoneType` if optional. [This is still done all the same](https://github.com/python/mypy/pull/12815/files#diff-c06d82e1af489188cdaa58e70e6460a858e2d4a993281e440a0c581947d69d81R657), but directly on parsing the converter, so this internal value does not have to be saved and passed (and [de]serialized).
Same thing with `is_invalid_converter`: before it was kept in the `Converter` info so that if it is True `AnyType(TypeOfAny.from_error)` can be given as the `init_type`, now we directly keep the appropriate `init_type`.

Reply:
Thanks, that clears it up.